### PR TITLE
#162163004 Rate article

### PR DESCRIPTION
**Pull request by katunold**
#### What does this PR do?
* Implement rate article feature
* Write tests for rate article
#### Description of Task to be completed?
- An authenticated user should be able to rate articles written by other authors on a scale of `0` to `5`
- Display articles with their average ratings 
#### How should this be manually tested?
After cloning the repo, cd into it and follow the steps 
* In terminal, type `git checkout ft-rate-article-162163004` to switch to my branch
* Open your prefered editor and go to the settings. py file in the project
* In the settings.py file, edit the value assigned to a password in Database to what you use locally on your computer
* Run the server using python manage.py runserver
* Copy this url `api/slug/rate/` 
* Open postman and paste the above route then replace the word slug with the slug of the article you want to rate.  **Note.** don't forget to include the host at the start of the route
* Access postman headers and feed in the following data:
_key_: content-type _value_: application/json
_key_: Authorization _value_: Token **your token**
* Access postman body and feed in your rating just like the example below
```
{
  "article": {
    "score": 4
  }
}
```
* Press send on postman and expect the following response
```
{
  "articles": {
    "score": "4.00",
    "rated_by": "Arnold2",
    "rated_on": "2018-12-11T03:08:48.584361Z",
    "article": "who-is-he",
    "average_rating": 4
  }
}
```
#### What are the relevant pivotal tracker stories?
[#162163004](https://www.pivotaltracker.com/story/show/162163004)

### CODE REVIEW

File: authors/apps/article/serializers.py [hunk 1]
```diff
@@ -31,7 +37,8 @@ class Meta:
         class behaviours
         """
         model = Article
-        fields = '__all__'
+        fields = ('slug', 'title', 'description', 'body', 'created_on', 'average_rating', 'user_rating',
```
Comment (billkabanga) [inline authors/apps/article/serializers.py]: Please try to follow PEP8 standards.

Comment (arthurarty) [reply authors/apps/article/serializers.py]: Hey did you run the yapf -i <file name> command.

Comment (katunold) [reply authors/apps/article/serializers.py]: I worked on the feedback you raised

**Review by billkabanga — APPROVED** [review]

File: authors/apps/tests/test_rating_article.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    def test_rate_someones_article(self):
+        """
+        method to test rating someone's article
+        :return:
+        """
+        self.authorize_user()
+        self.posting_article(post_article)
+
+        self.authorize_user2()
+        response = self.client.get(reverse("articles"), format='json')
```
Comment (malep2007) [inline authors/apps/tests/test_rating_article.py]: Could you handle this get request while initialising an object in this class? Ideally, it is clumsy to have two responses in the same method. It is something I have noticed throughout this test suite.

**Review by malep2007 — CHANGES_REQUESTED** [review]
Please work on handling the initial request in the init method of the class. It is a little clumsy to have two responses in the same test method.

File: authors/apps/tests/test_rating_article.py (hunk 2 above)
Comment (katunold) [reply authors/apps/tests/test_rating_article.py]: cool let me work on it

File: authors/apps/tests/base.py [hunk 3]
```diff
@@ -61,3 +61,20 @@ def deleter(self, slug):
...
+        self.authorize_user2()
+        response = self.client.get(reverse("articles"), format='json')
+        data = response.json().get("articles")[0]
+        slug = data["slug"]
+
+        return self.client.post(
+            reverse("rating", kwargs=dict(slug=slug)),
+            data={"article": {"score": rating}},
+            format="json"
+        )
```
Comment (malep2007) [inline authors/apps/tests/base.py:80]: Please add a new line

**Review by malep2007 — COMMENTED** [review]
Ok for now. In the future, please look out for these small things that make your code readable.

**Review by malep2007 — APPROVED** [review]

**Pull request merged**